addons/stock/models/stock_move_line.py

```diff
@@ -481,15 +489,23 @@ def _get_similar_move_lines(self):
         return lines
 
     def _create_and_assign_production_lot(self):
-        """ Creates and assign a new production lot of the move line."""
+        """ Creates and assign new production lots for move lines."""
+        mls = self.filtered(lambda ml: ml.lot_name and not ml.lot_id)
```

Comment:
do you need to filter here?i think ml_to_create_lot only contains relevant move lines

Reply:
@sle-odoo  Previously it works like this, because we didn't filter out ml already assigned a lot_id. I also think this is strange. I will move this part to ml.action_done().